core/unit_test/TestViewMapping_a.hpp

```diff
@@ -1009,9 +1009,6 @@ void test_view_mapping() {
     ASSERT_EQ(a.use_count(), 1);
     ASSERT_EQ(b.use_count(), 0);
 
-#if !defined(KOKKOS_ENABLE_CUDA) || !defined(KOKKOS_ENABLE_CUDA_LAMBDA)
-    // Cannot launch host lambda when CUDA lambda is enabled.
```

Comment:
https://github.com/kokkos/kokkos/pull/5628#discussion_r1094515352
Daniel will need to reexplain to me...

Reply:
Something funky with reference counting going on with lambdas. It's always `7` here. It works if you turn it into a functor...
```diff
diff --git a/core/unit_test/TestViewMapping_a.hpp b/core/unit_test/TestViewMapping_a.hpp
index 49edc16f8..83da1b7e2 100644
--- a/core/unit_test/TestViewMapping_a.hpp
+++ b/core/unit_test/TestViewMapping_a.hpp
@@ -24,6 +24,25 @@
 
 namespace Test {
 
+       template <typename Space>
+       struct DummyFunctor {
+
+Kokkos::View<int *, Space> a;
+
+               KOKKOS_FUNCTION void operator()(int i, int& e) const {
+         // for parallel_reduce we copy the functor into a combined
+          // functor-reducer object (with reference-counting on) before
+          // constructing the ParallelReduce object (with reference-counting
+          // turned off). When the parallel dispatch accepts a move for the
+          // lambda, this count should become 2.
+
+          if (a.use_count() != 3) { Kokkos::printf("1: %d should be 3\n", a.use_count()); ++e; }
+         Kokkos::View<int*, Space> x = a;
+          if (a.use_count() != 3) { Kokkos::printf("2: %d should be 3\n", a.use_count()); ++e; }
+          if (x.use_count() != 3) { Kokkos::printf("3: %d should be 3\n", a.use_count()); ++e; }
+               }
+       };
+
 template <class Space>
 void test_view_mapping() {
   using ExecSpace = typename Space::execution_space;
@@ -1014,7 +1033,7 @@ void test_view_mapping() {
 
     int errors = 0;
     Kokkos::parallel_reduce(
-        Kokkos::RangePolicy<host_exec_space>(0, 10),
+        Kokkos::RangePolicy<host_exec_space>(0, 10), DummyFunctor<Space>{a}, /*
         KOKKOS_LAMBDA(int, int& e) {
           // for parallel_reduce we copy the functor into a combined
           // functor-reducer object (with reference-counting on) before
@@ -1022,11 +1041,11 @@ void test_view_mapping() {
           // turned off). When the parallel dispatch accepts a move for the
           // lambda, this count should become 2.
 
-          if (a.use_count() != 3) ++e;
+          if (a.use_count() != 3) { Kokkos::printf("1: %d should be 3\n", a.use_count()); ++e; }
           V x = a;
-          if (a.use_count() != 3) ++e;
-          if (x.use_count() != 3) ++e;
-        },
+          if (a.use_count() != 3) { Kokkos::printf("2: %d should be 3\n", a.use_count()); ++e; }
+          if (x.use_count() != 3) { Kokkos::printf("3: %d should be 3\n", a.use_count()); ++e; }
+        },*/
         errors);
     ASSERT_EQ(errors, 0);
   }
```